### PR TITLE
WIP: More modular js files

### PR DESCRIPTION
**Pull request by djhvscf**

### CODE REVIEW

**Comment by djhvscf** [comment]
Ref: #4283

**Comment by djhvscf** [comment]
Ref: #4296

File: src/utils/index.js [hunk 1]
```diff
@@ -203,17 +198,17 @@ export default {
...
     const m = []
 
-    $els.each((y, el) => {
+    $els.forEach((el, y) => {
       const row = {}
 
       // save tr's id, class and data-* attributes
-      row._id = $(el).attr('id')
-      row._class = $(el).attr('class')
+      row._id = el.getAtrribute('id')
```
Comment (UtechtDustin) [inline src/utils/index.js]: typo `getAttribute`

File: src/utils/index.js [hunk 2]
```diff
@@ -203,17 +198,17 @@ export default {
...
+      row._class = el.getAtrribute('class')
       row._data = this.getRealDataAttr($(el).data())
 
-      $(el).find('>td,>th').each((_x, el) => {
-        const cspan = +$(el).attr('colspan') || 1
-        const rspan = +$(el).attr('rowspan') || 1
+      find(el, '>td,>th').forEach((el, _x) => {
```
Comment (UtechtDustin) [inline src/utils/index.js]: `'>td,>th'` is not a valid css selector

File: src/bootstrap-table.js [hunk 3]
```diff
@@ -134,27 +143,28 @@ class BootstrapTable {
...
     } else if (this.options.theadClasses) {
-      this.$header.addClass(this.options.theadClasses)
+      addClass(this.$header, this.options.theadClasses)
     }
-    this.$header.find('tr').each((i, el) => {
+    find(this.$header, 'tr').forEach((tr, i) => {
       const column = []
 
-      $(el).find('th').each((i, el) => {
+      find(tr, 'th').forEach((i, th) => {
```
Comment (UtechtDustin) [inline src/bootstrap-table.js]: wrong parameter order `th, i`

File: src/dom/dom.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
+import {isString, isUndefined, isJQueryObject} from '../utils/types'
```
Comment (UtechtDustin) [inline src/dom/dom.js]: missing `.js` at the end, without my browser cant load it!

Comment (UtechtDustin) [reply src/dom/dom.js]: same on other lines/files e.g. the line here under it

File: src/utils/index.js (hunk 1 above)
Comment (UtechtDustin) [reply src/utils/index.js]: A lot other lines are affect too

File: src/bootstrap-table.js [hunk 5]
```diff
@@ -307,50 +317,59 @@ class BootstrapTable {
       html.push('</tr>')
     })
 
-    this.$header.html(html.join(''))
-    this.$header.find('th[data-field]').each((i, el) => {
+    this.$header.innerHTML = ''
+    this.$header.appendChild(createElem(html.join('')))
```
Comment (UtechtDustin) [inline src/bootstrap-table.js]: missing `[0]` ?

**Comment by UtechtDustin** [comment]
Please update the PR to a version which works.
@wenzhixin and me determinated that we should remove jquery step by step, because its hard to review a large pull request.

Also please separate the bugs/feature, this branch e.g. contains https://github.com/wenzhixin/bootstrap-table/pull/4296. Two pull requests are better in mine/our opinion, that is the reason why git can rebase :)

But thank you for you great work, i would help but there are some issues in the branch so i'm not sure it the last version of your local branch ?

**Comment by djhvscf** [comment]
Hi, yes it is under development, there are parts of this PR which are not fully tested, that's why I have typos and wrong parameters order in forEachs..

You're right related to changes in filter control.. I can create a separate PR for that feature.

Remove the jQuery from main js file can't be done step by step since if an object is not a jQuery object this brokes everything.

File: src/utils/index.js (hunk 2 above)
Comment (djhvscf) [reply src/utils/index.js]: Agree

File: src/utils/index.js (hunk 1 above)
Comment (djhvscf) [reply src/utils/index.js]: Yes, it is an issue in latest commit

File: src/dom/dom.js (hunk 4 above)
Comment (djhvscf) [reply src/dom/dom.js]: Ok, if this can be added to eslint, it would be great

File: src/bootstrap-table.js (hunk 3 above)
Comment (djhvscf) [reply src/bootstrap-table.js]: Agree, issue in latest commit

File: src/bootstrap-table.js (hunk 5 above)
Comment (djhvscf) [reply src/bootstrap-table.js]: No here.

**Comment by UtechtDustin** [comment]
But it should be possible or not ?
Quote from @wenzhixin:
> I agree with you, for example replace some jQuery utils, replace query selector...

**Comment by djhvscf** [comment]
My main work on this pull request is related to change query selectors to vanilla js

So from my point of view this is not possible step by step at least in main .js file. But this just my opinion, please let me know if this needs to be postponed so I can work on different things.

**Comment by djhvscf** [comment]
@wenzhixin

**Comment by UtechtDustin** [comment]
Okay, then it should fine.
But we should preventing to refactor the whole jquery stuff in one Pull request! :)

**Comment by djhvscf** [comment]
Ok then, in that case the best option for us is close this PR and the changes in this PR can be submitted PR by PR.

**Comment by UtechtDustin** [comment]
Any news here ? @djhvscf

**Comment by djhvscf** [comment]
We can close this pull request

**Pull request closed without merging**